addons/auth_signup_recaptcha/controllers/signup.py

```diff
@@ -0,0 +1,18 @@
+from odoo import http, _
+from odoo.addons.auth_signup.controllers.main import AuthSignupHome
+from odoo.http import request
+from odoo.exceptions import UserError
+
+
+class SignupCaptcha(AuthSignupHome):
+    @http.route()
+    def web_auth_signup(self, *args, **kw):
+        if request.httprequest.method == 'POST' and not request.env['ir.http']._verify_request_recaptcha_token('signup'):
```

Comment:
How does it work then if you have turnstile without google_recaptcha ? Where does it verify the captcha token ?

Reply:
> website_cf_turnstile depends on google_recaptcha

You mean `auth_signup_turnstile` depends on `google_recaptcha` I guess, not `website_cf_turnstile`.

Isn't this a bit weird to force the dependence to google recaptcha when you install turnstile ? The common sense would say that you choose one or the other, not both, so enforcing the dependence seems weird in my opinion.

I also see that you add `website_cf_turnstile` as a dependence of `auth_sigup_turnstile`, meaning if you want turnstile for the signup/reset password, you force people to install the website. In addition to be weird, it means that you will force an upsell to customers who want the turnstile protection and who do not have/pay for the website module. As the pricing is done by module.

Meaning some customers will need to pay more in order to have the turnstile protection on signup / reset password if they do not have yet the website.

Reply:
`website_cr_turnstile` doesn't depends on `google_recaptcha` in its manifest

https://github.com/odoo/odoo/blob/07a0c98bb7be63d38f2c2148919f2ddd8bd370b6/addons/website_cf_turnstile/__manifest__.py#L11

The comment you mention doesn't make it a dependence. It just means that "if google_recaptacha" is installed, then we need to check its result first / as well. But it doesn't make it a dependence.

It just a series of method overrides `_verify_request_recaptcha_token`.

Reply:
Well then it's a bug because in the manifest this module doesn't depend on `google_recaptcha`. Don't you agree on this ?

Reply:
Ok I said nothing, it's website itself which depends on `google_recaptcha`.

I didn't see that coming